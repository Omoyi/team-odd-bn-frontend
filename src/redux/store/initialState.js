
const state = {
  search: {
    key: '',
    payload: '',
  },
  popUpsDisplay: {
    deleteComment: 'none',
    editTrip: 'none',
    currentPopUp: 'editTrip',
  },
  pagination: {
    pageNo: 0,
    itemsPerPage: 5,
  },
  accommodation: {
    createAccommodation: {
      rooms: [],
      photos: [],
    },
  },
  auth: {
    signup: {
      spinnerStatus: false,
      signupInputData: {},
      signupData: {},
      signupError: {},
    },
    signin: {
      spinnerStatus: false,
      signinInputData: {},
      signinData: {},
      signinError: {},
    },
  },
  allUsers: {
    paginatedRequest: [],
    paginationStart: 0,
    paginationEnd: 5,
  },
  trips: {
    tripRequests: {
      getCity: [],
      onewayData: {},
      onewayError: {},
      onewayInput: {},
    },
    availRequests: {
      approveRequestMessage: {},
    },
    requests: {
      requestsData: [],
      singleRequestData: {},
      requestCommentsData: {},
      requestsError: {},
      paginatedRequest: [],
      paginationStart: 0,
      paginationEnd: 5,
    },
    getStats: {
      getStatsInput: {
        tripType: 1,
      },
      getStatsData: {},
      getStatsError: {},
      tripStatsCounter: {},
    },
  },
  viewProfile: {
    profile: {},
  },
  requestsData: {
    data: '',
  },
};
export default state;
