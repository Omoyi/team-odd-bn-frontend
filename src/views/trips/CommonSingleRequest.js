import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { plusDivs } from '../../helpers/carousel';
import Action from '../../components/Action';
import verifyToken from '../../helpers/verifyToken';

class RequestView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isApproved: false,
      isRejected: false,
    };

    // this.handleChange = this.handleChange.bind(this);
  }

  handleChange(value) {
    this.setState({
      isApproved: !value,
      isRejected: value,
    });
  }

  render() {
    const { isApproved, isRejected } = this.state;
    console.log('state', this.state);
    const {
      comments, trips, approveRequest, params,
    } = this.props;
    this.refs.trips && plusDivs(1, this.refs.trips.children);

    return (
      <>
        <div className="singleRequestHeader">
          { verifyToken(localStorage.getItem('token')).roleId === 6
            ? (
              <>
                <div>
                  <button
                    id="approve"
                    disabled={isApproved}
                    className="approve-button"
                    onClick={() => {
                      this.handleChange(false);
                      approveRequest(params.tripRequestId, 'status=accept');
                    }}
                    background="#34c6f3"
                    type="submit"
                    action="approve"
                  >
                    {isApproved ? 'Approved' : 'Approve'}
                  </button>

                </div>
                <div>
                  <button
                    id="reject"
                    disabled={isRejected}
                    className="reject-button"
                    onClick={() => {
                      this.handleChange(true);
                      approveRequest(params.tripRequestId, 'status=reject');
                    }}
                    background="#34c6f3"
                    type="submit"
                    action="approve"
                  >
                    {isRejected ? 'Rejected' : 'Reject'}
                  </button>
                </div>
              </>
            )
            : (
              <div><Action background="#34c6f3" url="#" action="edit" /></div>
            )}
        </div>
        <div className="singleRequest">
          <div className="trips" ref="trips">
            <div><h3>Trip Details</h3></div>
            {trips && trips.trips
                && trips.trips.map((trip, index) => (
                  <div className="trip" key={index} ref={(index) => index}>
                    <div className="originId">{trip.originId}</div>
                    <div className="destinationId">{trip.destinationId}</div>
                    <div className="startDate">{trip.startDate.slice(0, 10)}</div>
                    <div className="returnDate">{trip.returnDate === null ? null : trip.returnDate.slice(0, 10)}</div>
                    <div className="reason">{trip.reason}</div>
                  </div>
                ))}
            <div className="corasselButtons">
              {verifyToken(localStorage.getItem('token')).roleId === 6 ? (
                <div>
                  <a className="back-button" href="/trips/approval">Back To Trips</a>
                </div>
              )
                : (
                  <div>
                    <a className="back-button" href="/requests">Back To Trips</a>

                  </div>
                )}
              <div>
                <button id="back" type="button" onClick={() => plusDivs(-1, this.refs.trips.children)}>
              &nbsp;
                </button>

              </div>
              <div>
                <button id="front" type="button" onClick={() => plusDivs(1, this.refs.trips.children)}>
              &nbsp;
                </button>
              </div>
            </div>
          </div>
          <div className="comments">
            <div><h3>comments</h3></div>
            {comments && comments.map((comment, index) => (
              <div className="commentCardContainer" key={index}>
                <div className="commentCardHeader">
                  <div className="commentCardHeaderLeftSide">
                    {comment.user.firstName}
                  </div>
                  <div className="commentCardHeaderRigthSide">
                    {comment.updatedAt.slice(0, 10)}
                  </div>
                </div>
                <div className="commentBody" key={comment.key}>
                  {comment.comment}
                </div>
              </div>
            ))}
            <div className="commentCardContainer">
              <div className="commentCardHeader">
                <div className="commentCardHeaderLeftSide">
                    Comment
                </div>
                <div className="commentCardHeaderRigthSide" />
              </div>
              <div className="commentInputField">
                <textarea />
              </div>
            </div>
            <div className="singleRequestFooter">
              <button type="button">post</button>
            </div>
          </div>
        </div>
      </>
    );
  }
}

export default RequestView;
