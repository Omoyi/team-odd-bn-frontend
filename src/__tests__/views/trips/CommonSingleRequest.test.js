import React from 'react';
import dotenv from 'dotenv';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import CommonSingleRequest from '../../../views/trips/CommonSingleRequest';
import initialState from '../../../redux/store/initialState';
import {
  init,
} from '../../../__mocks__/trips/requests';
import 'jest-localstorage-mock';

dotenv.config();
const mockStore = configureStore([]);
let wrapper;
let store;

describe('Common Single Request page ', () => {
  it('Test snapshot using the initial state', () => {
    store = mockStore(initialState);

    wrapper = shallow(
      <Provider store={store}>
        <Router>
          <CommonSingleRequest
            cities={init.trips.tripRequests.getCity}
            store={store}
            handleSubmit={async () => {
              jest.fn();
              return jest.fn();
            }}
            handleChange={(value) => {
              jest.fn();
              return jest.fn(value);
            }}
            postCommentsAction={jest.fn()}
          />
        </Router>

      </Provider>,

    );
    expect(wrapper).toMatchSnapshot();
  });
  it('Test snapshot using the initial state', () => {
    store = mockStore(init);

    wrapper = shallow(
      <Provider store={store}>
        <Router>
          <CommonSingleRequest
            comments={init.trips.requests.requestCommentsData}
            trips={init.trips.requests.singleRequestData}
            cities={init.trips.tripRequests.getCity}
            postCommentsAction={jest.fn()}
          />
        </Router>

      </Provider>,

    );
    expect(wrapper).toMatchSnapshot();
  });

  it('it Should dispatch hit submit button successfully', async () => {
    const plusDivs = jest.fn();
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <CommonSingleRequest
            comments={init.trips.requests.requestCommentsData}
            trips={init.trips.requests.singleRequestData}
            cities={init.trips.tripRequests.getCity}
            singleRequestData={init.trips.requests.singleRequestData}
            params={{ tripRequestId: 8 }}
            approveRequest={jest.fn()}
            postCommentsAction={jest.fn()}
            updateCommentInputAction={jest.fn()}
          />
        </Router>
      </Provider>,
    );
  });

  it('it Should dispatch reject button successfully', async () => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <CommonSingleRequest
            comments={init.trips.requests.requestCommentsData}
            trips={init.trips.requests.singleRequestData}
            cities={init.trips.tripRequests.getCity}
            params={{ tripRequestId: 8 }}
            approveRequest={jest.fn()}
            postCommentsAction={jest.fn()}
            updateCommentInputAction={jest.fn()}
          />
        </Router>
      </Provider>,
    );
    const postComment = wrapper.find('#postComment');
    const submitComment = wrapper.find('#submitComment');
    const deleteComment = wrapper.find('#deleteComment');
    const edit = wrapper.find('#edit');
    // const approve = wrapper.find('#approve');
    // const reject = wrapper.find('#reject');
    const page1 = wrapper.find('#page1S');
    // const page2 = wrapper.find('#page2S');
    const page3 = wrapper.find('#page3S');
    page1.simulate('click');
    // page2.simulate('click');
    page3.simulate('click');
    postComment.simulate('change');
    submitComment.simulate('click');
    deleteComment.simulate('click');
    edit.simulate('change');
    expect(jest.fn().mock.calls).toBeDefined();
  });
  it('it Should dispatch reject button successfully', async () => {
    store = mockStore(init);
    localStorage.setItem('token', process.env.TESTING_TOKEN);
    console.log('{{{{}}}}{{{{{{', init.trips.tripRequests.getCity);

    wrapper = mount(
      <Provider store={store}>
        <Router>
          <CommonSingleRequest
            comments={init.trips.requests.requestCommentsData}
            cities={init.trips.tripRequests.getCity}
            trips={init.trips.requests.singleRequestData}
            params={{ tripRequestId: 8 }}
            approveRequest={jest.fn()}
            postCommentsAction={jest.fn()}
            updateCommentInputAction={jest.fn()}
          />
        </Router>
      </Provider>,
    );
    const approve = wrapper.find('#approve');
    const reject = wrapper.find('#reject');
    approve.simulate('click');
    reject.simulate('click');
    expect(jest.fn().mock.calls).toBeDefined();
  });
});
