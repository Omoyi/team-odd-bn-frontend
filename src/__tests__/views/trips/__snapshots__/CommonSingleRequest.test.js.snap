// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Single Request page  Test snapshot using the initial state 1`] = `ShallowWrapper {}`;

exports[`Common Single Request page  Test snapshot using the initial state 2`] = `ShallowWrapper {}`;
